agent/engine/task_manager_test.go

```diff
@@ -201,8 +201,61 @@ func TestContainerNextState(t *testing.T) {
 			assert.Equal(t, tc.expectedContainerStatus, nextStatus,
 				"Expected next state [%s] != Retrieved next state [%s]",
 				tc.expectedContainerStatus.String(), nextStatus.String())
-			assert.Equal(t, tc.expectedTransitionActionable, actionRequired)
-			assert.Equal(t, tc.expectedTransitionPossible, possible)
+			assert.Equal(t, tc.expectedTransitionActionable, actionRequired, "transition actionable")
+			assert.Equal(t, tc.expectedTransitionPossible, possible, "transition possible")
+		})
+	}
+}
+
+func TestContainerNextStateWithDependencies(t *testing.T) {
+	testCases := []struct {
+		containerCurrentStatus       api.ContainerStatus
+		containerDesiredStatus       api.ContainerStatus
+		dependencyCurrentStatus      api.ContainerStatus
+		expectedContainerStatus      api.ContainerStatus
+		expectedTransitionActionable bool
+		expectedTransitionPossible   bool
+	}{
+		// NONE -> RUNNING transition is not allowed and not actionable, when desired is Running and dependency is None
+		{api.ContainerStatusNone, api.ContainerRunning, api.ContainerStatusNone, api.ContainerStatusNone, false, false},
+		// NONE -> RUNNING transition is not allowed and not actionable, when desired is Running and dependency is Created
+		{api.ContainerStatusNone, api.ContainerRunning, api.ContainerCreated, api.ContainerStatusNone, false, false},
+		// NONE -> RUNNING transition is allowed and actionable, when desired is Running and dependency is Running
+		// The expected next status is Pulled
+		{api.ContainerStatusNone, api.ContainerRunning, api.ContainerRunning, api.ContainerPulled, true, true},
+		// NONE -> RUNNING transition is allowed and actionable, when desired is Running and dependency is Stopped
+		// The expected next status is Pulled
+		{api.ContainerStatusNone, api.ContainerRunning, api.ContainerStopped, api.ContainerPulled, true, true},
+	}
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("%s to %s Transition",
+			tc.containerCurrentStatus.String(), tc.containerDesiredStatus.String()), func(t *testing.T) {
```

Comment:
is .String() implied by %s? Or am I thinking too much in Java?

Reply:
wait, what's the question?

Reply:
`.String()` is implied: https://play.golang.org/p/KpcJEdaPKy

I was following the existing code.  Can change it if you want.

Reply:
I'm not sure what the convention is, but I'll typically vote for less code. Up to you.